BE/src/questions/questions.service.ts

```diff
@@ -66,6 +66,36 @@ export class QuestionsService {
     };
   }
 
+  async deleteOneQuestion(id: number, userId: number): Promise<boolean> {
+    const question = await this.prisma.question.findUnique({
+      where: {
+        Id: id,
+        IsAdopted: false,
+      },
+      include: {
+        User: {
+          select: {
+            Id: true,
+          },
+        },
+      },
+    });
+
+    if (!question) {
+      return false;
+    }
+
+    if (question.User.Id !== userId) {
+      throw new Error('User is not the owner of the question');
+    }
+
+    await this.prisma.question.delete({
+      where: {
+        Id: id,
+      },
+    });
+
+    return true;
```

Comment:
이 부분이 조금 고민되네요. 저도 유사하게 구현했을 것 같긴합니다.

고민되는 부분이 두가지가 있습니다.
1. findUnique를 실행하고 delete를 실행하는 사이에 모종의 이유로(다른 클라이언트에서 삭제 등) row가 삭제될 일은 없을까? 이런 부분을 고려해야할까?

2. delete문을 그냥 실행하고 몇 개 쿼리를 지웠는지 확인만해도 되지 않을까 고민입니다.
```suggestion
  async deleteOneQuestion(id: number, userId: number): Promise<boolean> {
    const result = await this.prisma.question.delete({
      where: {
        Id: id,
        IsAdopted: false,
        User: {
          Id: userId,
        },
      },
    });

    if (!result.count) {
      throw new NotFoundException('Question not found or user is not the owner');
    }

    return true;
```

물론 이런 방식으로 하면 어떤 이유로 row가 삭제되었는지 명확히 알 수 없는 단점도 존재합니다.

태호님 생각은 어떤가요?